packages/server/lib/controllers/v1/account/signin.ts

```diff
@@ -0,0 +1,54 @@
+import { z } from 'zod';
+import { asyncWrapper } from '../../../utils/asyncWrapper.js';
+import { requireEmptyQuery, zodErrorToHTTP } from '@nangohq/utils';
+import { getUserFromSession } from '../../../utils/utils.js';
+import { sendVerificationEmail } from '../../../helpers/email.js';
+import type { WebUser, Signin } from '@nangohq/types';
+
+const validation = z
+    .object({
+        email: z.string().email(),
+        password: z.string().min(8)
+    })
+    .strict();
+
+export const signin = asyncWrapper<Signin>(async (req, res) => {
```

Comment:
Because of how the auth work, you are logged at this point if the email/password is correct, so any error should logout or we should refactor the login or move this logic in the auth middleware

Reply:
Not totally sure what you mean here? Can you add more detail / explain the scenario or issues with this signin flow.

Reply:
when you reach this function, you have a session wether your email is verified or not. So even with an error, you will still receive a valid cookie and can continue to navigate the dashboard

Reply:
https://github.com/NangoHQ/nango/assets/1637651/934d2da8-4521-453e-9fc6-547059220559

Reply:
Nice catch! Updated with https://github.com/NangoHQ/nango/pull/2173/commits/83cce999b36b7207d54369629745d193d657e055